CMakeLists.txt

```diff
@@ -435,7 +436,7 @@ endif()
 install(FILES ${CMAKE_SOURCE_DIR}/LICENSE.TXT
         DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/doc/${PROJECT_NAME}/")
 
-install(FILES examples/basic/basic.c
+install(FILES examples/basic/gpu_shared_memory.c examples/basic/basic.c
```

Comment:
I have a question about UMF examples installation. As I can see now we installing only source files, but I believe we should install corresponding cmake files t build examples.

Reply:
I'm not sure about cmakes... IMO the lightest example the better. I would even prefer adding single-line commnt to the file with options to gcc for building
@PatKamin ?

Reply:
From the customer point of view I prefer having build infrastructure installed together with sources. So that customers can just run regular `cmake` and `make` commands to build examples.

Reply:
Whether the examples are built or not depends on several CMake options, like [these two](https://github.com/oneapi-src/unified-memory-framework/blob/main/examples/CMakeLists.txt#L5) and all of [these](https://github.com/oneapi-src/unified-memory-framework/pull/384/files#diff-940e5356fa1137e646ea135a6f68cacbfad4fe7124c3a69163468f588acf9283R36-R38). To simplify the installation procedure and testing the installation of UMF we opted for installing all files always so the user can have these examples as a reference of some sample UMF usage.

If we were to add examples' directory CMake files (most probably modified to not include src/utils files ie.) to allow the user to build examples, we would have to do it only when given options were set when the UMF was built so that all required libs can be found. We would have to track in our installation tests in which CMake options set build files shall be installed and in which not.

Given that some libraries have to be installed for each example, I think this could be confusing for the user to see all the example files but being able to build some of them or even none of them.

Reply:
I would like to keep things simple.. If we want to add cmakes they should be really, really simple - without ifs, flags etc. But I'm not sure if having cmake for single-file examples is necessary

Reply:
I think we could have CMake's installed along with examples' source files, even if examples are not built. We could make some simple CMake's, even for a single-file examples.

That said, we could do that in a sep. PR, to change this behaviour for all examples at once.

Reply:
I filed an issue for that: #427